src/abstract/Collection.ts

```diff
@@ -1,6 +1,35 @@
-import Backbone from 'backbone';
+import Backbone, { AddOptions } from 'backbone';
+import { isArray, isObject, isUndefined } from 'underscore';
 import Model from './Model';
 
-export default class Collection<
-  TModel extends Model = Model
-> extends Backbone.Collection<TModel> {}
+type Module<TModel extends Model> = TModel extends Model<infer M> ? M : unknown;
+type ModelConstructor<TModel extends Model> = { new (mod: Module<TModel>, attr: any): TModel };
+
+export default class Collection<TModel extends Model = Model> extends Backbone.Collection<TModel> {
+  module!: Module<TModel>;
+  private newModel!: ModelConstructor<TModel>;
+
+  add(model: Array<Record<string, any>> | TModel, options?: AddOptions): TModel;
+  add(models: Array<Array<Record<string, any>> | TModel>, options?: AddOptions): TModel[];
+  add(model?: unknown, options?: AddOptions): any {
+    //Note: the undefined case needed because backbonejs not handle the reset() correctly
+    var models = isArray(model) ? model : !isUndefined(model) ? [model] : undefined;
+
+    models = models?.map(m => (m instanceof this.newModel ? m : new this.newModel(this.module, m))) ?? [undefined];
```

Comment:
IDK why but when I try to load the editor with these changes it seems to break at this line
![Schermata 2022-07-15 alle 13 49 12](https://user-images.githubusercontent.com/11614725/179217691-822368b6-3161-4553-8f46-974c5bf9dcf0.png)
Looks like `this.newModel` is undefined

Reply:
I think you forgot to update the dependency. Sorry forgot to write it in the description (Updated)

Reply:
Yeah, you're right, after update, it seems to work properly 👍